docs: Refine security documentation for clarity

Dor-bl opened the pull request:
## Proposed changes

Corrected formatting and phrasing for clarity and consistency throughout the document. Adjusted security feature descriptions and command examples.

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [Contributing Guide](https://appium.io/docs/en/latest/contributing/)
- [x] I have signed the CLA
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

Dor-bl committed:
Refine security documentation for clarity

Corrected formatting and phrasing for clarity and consistency throughout the document. Adjusted security feature descriptions and command examples.

eglitise commented on packages/appium/docs/en/guides/security.md:
```diff
@@ -53,10 +53,10 @@ might use. Here is an incomplete list of examples from some of Appium's official
-Some insecure features operate on the server level, and do not require a driver session. Enabling
-these features requires using the wildcard prefix:
+Some insecure features operate on the server level and do not require a driver session. Enabling
+These features require using the wildcard prefix:
```
Same here

eglitise commented on packages/appium/docs/en/guides/security.md:
````diff
@@ -73,17 +73,31 @@ appium --allow-insecure=first:foo
+Turn on all insecure features except one:
+
+```bash
+appium '--allow-insecure=*:*' --deny-insecure='*:foo'
````
This is the same as the example on line 88, since `--allow-insecure=*:*` is equivalent to `--relaxed-security`

Dor-bl committed:
Fix typos in security documentation

Dor-bl committed:
Update security.md for multiple insecure features

Clarified section on enabling multiple insecure features for drivers.

Dor-bl committed:
Update security.md to remove insecure feature examples

Removed examples of insecure feature commands.

eglitise commented on packages/appium/docs/en/guides/security.md:
````diff
@@ -73,17 +73,25 @@ appium --allow-insecure=first:foo
+Turn on multiple insecure features for specific drivers:
+
+```bash
+appium '--allow-insecure=uiautomator2:adb_shell,xcuitest:get_server_logs'
````
You can omit the extra single quotes since the asterisk symbol is not used

Dor-bl committed:
Remove section on multiple insecure features

Removed the section on enabling multiple insecure features for specific drivers.

Dor-bl committed:
Fix formatting of appium command examples in security.md

Dor-bl commented on packages/appium/docs/en/guides/security.md:
````diff
@@ -73,17 +73,31 @@ appium --allow-insecure=first:foo
 
 ```bash
-appium --relaxed-security --deny-insecure=*:foo
+appium --relaxed-security --deny-insecure='*:foo'
````
Done
